### PR TITLE
Move signal processing functions and sigaction calls to signals from redshift.c

### PR DESCRIPTION
**Pull request by maandree**

### CODE REVIEW

File: src/signals.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	   (hooks) to be reaped automatically. */
+	sigact.sa_handler = SIG_IGN;
+	sigact.sa_mask = sigset;
+	sigact.sa_flags = 0;
+
+	r = sigaction(SIGCHLD, &sigact, NULL);
+	if (r < 0) {
+		perror("sigaction");
+		return -1;
+	}
```
Comment (jonls) [inline src/signals.c]: You appear to be missing a `return` statement here.

File: src/signals.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#ifndef REDSHIFT_SIGNALS_H
```
Comment (jonls) [inline src/signals.h]: Can you add the license header to these files? These can just be copied from redshift.c and changed to the right file name/author/date.

File: src/signals.c [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+sigdisable(int signo)
+{
+	disable = 1;
+}
+
+#endif /* ! HAVE_SIGNAL_H || __WIN32__ */
+
+
+int
+install_signal_handlers()
```
Comment (jonls) [inline src/signals.c]: should be `install_signal_handlers(void)`.

**Comment by jonls** [comment]
Thanks, this seems like a very reasonable part of the code to move out to a separate file :+1:

**Comment by maandree** [comment]
I have fixed to issued you noted.

**Comment by maandree** [comment]
Rebased too.

**Comment by jonls** [comment]
Thanks! I would like to prefix the functions and global variables with the name of the file. What I mean is that the function should be called `signals_install_handlers` and the global variables should be `signals_exiting` and `signals_disable`. This should make it more obvious in `redshift.c` that these symbols are defined in `symbols.c`. What do you think?

**Pull request closed without merging**